### PR TITLE
Move all classes into Pennyworth module

### PR DESCRIPTION
**Pull request by imobach**
Hi,

This PR move all Pennyworth classes to its own namespace (module `Pennyworth`). I'm doing that for a couple of reasons:

* It's the preferred way to organize Ruby applications and libraries :)
* It reduces the chances of class name clobbering.
* It makes Pennyworth to play nicely with Rubygems (and even Bundler). For example, if you want to use the Pennyworth RSpec helpers, you don't have to know what is the installation directory of Pennyworth anymore. You just have to type `require "pennyworth/spec"` (we are using them in https://github.com/schubi2/autoyast_test).
* If you're using bundler and Pennyworth is a dependency of your application, you can run the main script just calling `bundle exec pennyworth setup` (without known anything about Pennyworth path).

The PR has a lot of changes just because I have to change the indentation for most files. I've run the test suite and it seems to be working right now. Also it is working for us as a dependency for autoyast_test.

Thank you in advance.

Regards,
Imo

### CODE REVIEW

**Comment by imobach** [comment]
The idea is that this PR is only to move classes into Pennyworth module. So I guess that I'd must have the rest of the code untouched (ignoring houndci warnings for now).

**Comment by imobach** [comment]
It seems like last commits on master make this PR impossible to merge at this time. I'll rebase it.

**Comment by imobach** [comment]
Hi all,

I've just rebased the PR to master.

Regards,
Imobach

**Comment by thardeck** [comment]
Hey,

thx for your pr. We will look into it on Monday.

Regards,
Tim

File: lib/pennyworth.rb [hunk 1]
```diff
@@ -28,35 +28,38 @@
...
-require_relative "cli_host_controller"
-require_relative "host_config"
-require_relative "lock_service"
-require_relative "remote_command_runner"
-require_relative "local_runner"
-require_relative "local_command_runner"
+module Pennyworth
+end
+
+require "pennyworth/version"
```
Comment (aduffeck) [inline lib/pennyworth.rb]: This change breaks loading the pennyworth library locally without being pulled in as a gem, so I think using `require_relative` makes sense here.

Comment (imobach) [inline lib/pennyworth.rb]: Ok, I didn't think about that. Fixed.

File: lib/pennyworth/spec.rb [hunk 2]
```diff
@@ -21,23 +21,23 @@
...
-require_relative "host_runner"
-require_relative "lock_service"
-require_relative "ssh_keys_importer"
-require_relative "vm"
-require_relative "spec_profiler"
-require_relative "remote_command_runner"
-require_relative "settings"
-require_relative "local_runner"
-require_relative "local_command_runner"
+require "pennyworth/exceptions"
```
Comment (aduffeck) [inline lib/pennyworth/spec.rb]: I'm sorry but I realized that we actually need the require_relative change here, not in `pennyworth.rb` (although it doesn't hurt there either, so I'd keep it).

**Comment by aduffeck** [comment]
lgtm besides the other `require_relative` change. Once that's changed we can merge this pr.

Thanks :)

**Comment by imobach** [comment]
Thanks @aduffeck :)

I've changed the rest of requires and I've also noticed of a problem with the path to the `setup.yml` file for `Pennyworth::Command` class. It should be fixed now.

**Comment by imobach** [comment]
I've rebased the changes again.

**Comment by aduffeck** [comment]
Merged, thanks for your efforts :)

**Comment by imobach** [comment]
Thanks a lot!

**Pull request merged**